packages/util/src/signature.ts

```diff
@@ -175,14 +188,15 @@ export const isValidSignature = function (
     return false
   }
 
-  const rBN = new BN(r)
-  const sBN = new BN(s)
+  const rBN = bufferToBigint(r)
+  const sBN = bufferToBigint(s)
+  const _0n = BigInt(0)
 
-  if (rBN.isZero() || rBN.gt(SECP256K1_N) || sBN.isZero() || sBN.gt(SECP256K1_N)) {
```

Comment:
I think there was an error in the logic. 

`s.gt(n)` is `s > n`. It must be `s >= n`.

Reply:
oh, nice find, do you think you could add some test cases to `signature.spec.ts` that covers this discrepancy?

I'm a little surprised it wasn't in [`ethereum/tests`](https://github.com/ethereum/tests), might be a good suggestion for over there too, boundaries can be problematic and always good to be well tested.

Reply:
(also, @paulmillr if you confirm this is a bug, we should also plan to fix on master and include as part of a util v7.x patch release)